chainer/functions/connection/linear.py

```diff
@@ -69,6 +69,81 @@ def backward(self, indexes, grad_outputs):
         return ret
 
 
+class LinearGradData(function_node.FunctionNode):
+
+    def __init__(self, linear):
+        return
+
+    def forward(self, inputs):
+        self.retain_inputs((0, 1))
+        gy, W = inputs
+
+        if not type_check.same_types(*inputs):
+            raise ValueError('numpy and cupy must not be used together\n'
+                             'type(gy): {0}, type(W): {1}'
+                             .format(type(gy), type(W)))
+        if (isinstance(gy, numpy.ndarray) and
+                not (gy.flags.c_contiguous or gy.flags.f_contiguous) and
+                1 in gy.shape):
+            gy = numpy.ascontiguousarray(gy)
+
+        gx = gy.dot(W).astype(gy.dtype, copy=False)
+        self.retain_inputs((0, 1))
```

Comment:
Inputs are already retained above.
(same for `LinearGradWeight`)

Reply:
I will remove it, thanks